### PR TITLE
WX: make Netplay use new-style config

### PR DESCRIPTION
**Pull request by ligfx**

### CODE REVIEW

File: Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp [hunk 1]
```diff
@@ -92,32 +92,22 @@ bool NetPlayLauncher::Join(const NetPlayJoinConfig& config)
...
 {
-  std::string host;
-
-  netplay_section.Get("TraversalServer", &host, DEFAULT_TRAVERSAL_HOST);
-  host = StripSpaces(host);
+  std::string host = StripSpaces(Config::Get(Config::NETPLAY_TRAVERSAL_SERVER));
 
   if (host.empty())
-    return DEFAULT_TRAVERSAL_HOST;
+    return Config::NETPLAY_DEFAULT_TRAVERSAL_HOST;
```
Comment (leoetlino) [inline Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp]: Can this be the default value for Config::NETPLAY_TRAVERSAL_SERVER? (I think the StripSpaces is also unnecessary as IniFile will do that.)

File: Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp [hunk 2]
```diff
@@ -92,32 +92,22 @@ bool NetPlayLauncher::Join(const NetPlayJoinConfig& config)
...
 
-u16 NetPlayLaunchConfig::GetTraversalPortFromIniConfig(const IniFile::Section& netplay_section)
+u16 NetPlayLaunchConfig::GetTraversalPort()
 {
-  std::string port_str;
-  unsigned long port;
-
-  netplay_section.Get("TraversalPort", &port_str, std::to_string(DEFAULT_TRAVERSAL_PORT));
-  StrToWxStr(port_str).ToULong(&port);
+  u16 port = Config::Get(Config::NETPLAY_TRAVERSAL_PORT);
```
Comment (leoetlino) [inline Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp]: Same question: could NETPLAY_DEFAULT_TRAVERSAL_PORT be set as the default value for this config info?

File: Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp [hunk 3]
```diff
@@ -92,32 +92,22 @@ bool NetPlayLauncher::Join(const NetPlayJoinConfig& config)
...
 
   if (port == 0)
-    port = DEFAULT_TRAVERSAL_PORT;
+    port = Config::NETPLAY_DEFAULT_TRAVERSAL_PORT;
 
   return static_cast<u16>(port);
```
Comment (leoetlino) [inline Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp]: The static_cast shouldn't be required anymore, as `port` is already a u16.

Comment (ligfx) [inline Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp]: Doh, thanks!

File: Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp (hunk 1 above)
Comment (ligfx) [inline Source/Core/DolphinWX/NetPlay/NetPlayLauncher.cpp]: Removed `StripSpaces`.

**Comment by ligfx** [comment]
@leoetlino On the topic of the default values, yeah, I was a bit unclear on that. It looks like there's no actual code to set it to something other than the default (unless someone's modifying INI files by hand), so there's no point in having empty defaults. Will change.

**Pull request merged**